Mathlib/Data/List/Dedup.lean

```diff
@@ -132,6 +133,41 @@ theorem dedup_append (l₁ l₂ : List α) : dedup (l₁ ++ l₂) = l₁ ∪ ded
   · rw [dedup_cons_of_not_mem' h, insert_of_not_mem h]
 #align list.dedup_append List.dedup_append
 
+theorem dedup_map_of_injective [DecidableEq β] {f : α → β} (hf : Function.Injective f)
+    (xs : List α) :
+    (xs.map f).dedup = xs.dedup.map f := by
+  induction xs with
+  | nil => simp
+  | cons x xs ih =>
+    rw [map_cons]
+    by_cases h : x ∈ xs
+    · rw [dedup_cons_of_mem h, dedup_cons_of_mem (mem_map_of_mem f h), ih]
+    · rw [dedup_cons_of_not_mem h, dedup_cons_of_not_mem <| (mem_map_of_injective hf).not.mpr h, ih,
+        map_cons]
+
+theorem Subset.dedup_append {xs ys : List α} (h : xs ⊆ ys) :
```

Comment:
```suggestion
theorem Subset.dedup_append_left {xs ys : List α} (h : xs ⊆ ys) :
```
and can we have the right version stated using `List.Perm`?

Reply:
Hmm maybe, but the fact it's missing compared to the other direction is quite upsetting and might cause confusion in the future. At least add a comment explaining why the other lemma doesn't exist

Reply:
I added the lemma, but in the multiset file where the proof was trivial.